tooling/specs/specs.go

```diff
@@ -0,0 +1,72 @@
+package specs
+
+import "fmt"
+
+type maturity string
+
+const (
+	wip        maturity = "wip"
+	draft      maturity = "draft"
+	reliable   maturity = "reliable"
+	stable     maturity = "stable"
+	permanent  maturity = "permanent"
+	deprecated maturity = "deprecated"
+)
+
+func (m maturity) isMature() bool {
+	switch m {
+	case reliable, stable, permanent:
+		return true
+	default:
+		return false
+	}
+}
+
+type Spec string
+
+const (
+	SubdomainGateway Spec = "subdomain-gateway"
+)
+
+var specMaturity = map[Spec]maturity{
+	SubdomainGateway: stable,
+}
+
+func (s Spec) IsMature() bool {
+	return specMaturity[s].isMature()
+}
+
+var specEnabled = map[Spec]bool{}
+
+func (s Spec) IsEnabled() bool {
+	if enabled, ok := specEnabled[s]; ok {
+		return enabled
+	} else {
+		return s.IsMature()
+	}
+}
```

Comment:
If a spec wasn't enabled/disabled explicitly, we infer whether it's enabled or not based on its' maturity.